client/landing/stepper/index.tsx

```diff
@@ -95,30 +95,26 @@ const FlowSwitch: React.FC< { user: UserStore.CurrentUser | undefined } > = ( {
 	const flowNameFromParam = useQuery().get( 'flow' );
 	const flowNameFromPathName = location.pathname.split( '/' )[ 1 ];
 	const recurType = useQuery().get( 'recur' );
-	let flow = siteSetupFlow;
 
 	// keep supporting the `flow` query param for backwards compatibility
 	if ( availableFlows.find( ( flow ) => flow.flowName === flowNameFromParam ) ) {
 		return <Redirect to={ `/${ flowNameFromParam }` } />;
 	}
 
+	let flow =
+		availableFlows.find( ( f ) => f.flowName === flowNameFromPathName )?.pathToFlow ||
+		siteSetupFlow;
 	if ( anchorFmPodcastId ) {
 		flow = anchorFmFlow;
-	} else if ( flowNameFromPathName === ecommerceFlow.name ) {
-		flow = ecommerceFlow;
+	}
+	if ( flow?.name === ecommerceFlow.name ) {
```

Comment:
Good catch on this subtle issue, @blackjackkent!

As a more general (and very definitely non-blocking!) note, I wonder whether this logic to pull in additional query parameters should be delegated out to the flow logic. The more flow-specific logic we handle in this block of code, the more complex this code block will become over time. We may want to spend some time thinking about whether it's worth pushing any data extraction from the query parameters into the flow, rather than having the general stepper code responsible for these parameters. The tradeoff will be that any URL query logic will be handled by the flow and won't be easy to track down from the central code.

Maybe not something to tackle immediately, but it may be worth dropping a comment here for future use cases so we can at least encourage our second use case to consider pushing this code into the flows themselves.

Reply:
This is a good point. I honestly don't think that this part of the code is a bad place for this logic but I agree about its potential to get bloated. I was wary of overengineering too heavily in getting this feature out the door at present, but I suspect the ultimate correct place for this logic would be in a self-contained hook that would then be called from within `FlowSwitch`.

Reply:
(i.e. `useFlowParams()` or something like that.)

Reply:
I agree with Dale here. I don't see why Stepper (as a framework) needs to be aware of the flow-specific query params. Particularly in `FlowSwitch` component that is no longer a switch after this change.